app/presenters/gui_presenter.rb

```diff
@@ -0,0 +1,29 @@
+class GuiPresenter
+
+  def read_gui_yaml
+    puts "READING YML"
+    yaml = YAML.load_file('resources/guis.yml')
+    return yaml["guis"]
+  end
+
+  def initialize
+    @guis_info = read_gui_yaml
+  end
+
+  @@instance = GuiPresenter.new
+
+  def self.instance
+    return @@instance
+  end
+
```

Comment:
@peff just pushed a new commit that uses `Rails.env.production?` to check if it should re-read the gui clients info at every request (the idea/hack is to ease developers doing changes on that part). I do think it's a bit hacky and dangerous, but I think it works for our use case. If you disagree we can revert the commit and try to find a solution (any other idea?).

Besides this issue, is anything else blocking merging this code PR that I can improve? :)

Reply:
It does feel hacky, but I think it will do the right thing in practice.

I tested this on the heroku site and it seems to work fine. I'll drop that `puts` myself and then merge.

Reply:
thanks!